### PR TITLE
Move to new k6 module API

### PR DESCRIPTION
**Pull request by MStoykov**
see https://github.com/grafana/k6/issues/2344 for more info

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -182,25 +181,29 @@ func (c *Client) sendQuery(ctx context.Context, q *Query) (httpext.Response, err
...
+		state.Logger.Warn("method pushParametrized() is deprecated and will be removed in future releases; please use pushParameterized() instead")
+	}
+	return c.PushParameterized(streams, minBatchSize, maxBatchSize)
 }
 
-func (c *Client) PushParameterized(ctx context.Context, streams, minBatchSize, maxBatchSize int) (httpext.Response, error) {
-	batch := newBatch(ctx, c.cfg.Labels, streams, minBatchSize, maxBatchSize)
-	return c.pushBatch(ctx, batch)
+func (c *Client) PushParameterized(streams, minBatchSize, maxBatchSize int) (httpext.Response, error) {
+	batch := newBatch(c.vu.Context(), c.vu.State(), c.cfg.Labels, streams, minBatchSize, maxBatchSize)
```
Comment (chaudum) [inline client.go:201]: I think we need to check for vu.State() here as well:

```suggestion
    if state := c.vu.State(); state == nil {
		return *httpext.NewResponse(), errors.New("state is nil")
	}
	batch := newBatch(c.vu.Context(), c.vu.State(), c.cfg.Labels, streams, minBatchSize, maxBatchSize)
```

**Review by chaudum — COMMENTED** [review]
One question, but otherwise LGTM. Still need to test it locally.

File: client.go (hunk 1 above)
Comment (MStoykov) [reply client.go:201]: I also wondered but it's already done in `pushBatch` so ... I don't think it's needed, but I can add it :shrug:

Comment (MStoykov) [reply client.go:201]: Do you want me to add it so we can merge this ?

Comment (chaudum) [reply client.go:201]: `PushParameterized` is the function that is usually called from within the JS test file. So I'd add the check there as well.

Comment (MStoykov) [reply client.go:211]: fixed in [77cd232](https://github.com/grafana/xk6-loki/pull/28/commits/77cd23223811a68a7f3025bf832bcc549755ec5d)

File: client.go [hunk 2]
```diff
@@ -189,6 +189,11 @@ func (c *Client) Push() (httpext.Response, error) {
 
 // PushParametrized is deprecated in favor or PushParameterized
 func (c *Client) PushParametrized(streams, minBatchSize, maxBatchSize int) (httpext.Response, error) {
+	state := c.vu.State()
+	if state == nil {
+		return *httpext.NewResponse(), errors.New("state is nil")
+	}
+
```
Comment (chaudum) [inline client.go:196]: You already added the check to this function in the lines below ;-)

**Review by chaudum — APPROVED** [review]

**Comment by MStoykov** [comment]
I was intending on using the github feature but I see it is disabled :(

**Comment by chaudum** [comment]
> I was intending on using the github feature but I see it is disabled :(

@MStoykov I enabled "squash merge" :)

**Pull request merged**